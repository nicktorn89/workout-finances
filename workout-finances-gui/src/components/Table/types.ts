import { WorkoutObject } from 'src/store/modules/types';
import { MouseEvent as ReactMouseEvent } from 'react';

export interface TableProps {
  data: WorkoutObject[];
  onCheckboxChange: (e: ReactMouseEvent<HTMLElement, MouseEvent>) => void;
}
