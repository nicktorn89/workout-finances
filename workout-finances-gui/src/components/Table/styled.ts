import { Table, Paper } from '@material-ui/core';
import styled from 'styled-components';

// @ts-nocheck
export const MaterialTable: any = styled(Table)`
  width: 80%;
  .table-head-cell {
    font-size: 1.25rem;
  }

  .table-body-cell {
    font-size: 1rem;
  }
`;

export const MaterialPaper: any = styled(Paper)`
  width: 80%;
  margin: 2em auto 0;
`;
