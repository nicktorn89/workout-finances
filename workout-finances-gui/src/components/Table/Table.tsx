import React, { memo } from 'react';
import { TableProps } from './types';
import { TableBody, TableHead, TableCell, TableRow } from '@material-ui/core';
import Done from '@material-ui/icons/Done';
import { MaterialTable, MaterialPaper } from './styled';

let id = 0;

function createData(date: Date, people: number, salary: number,
isFree: boolean, isPersonal: boolean, isJumps: boolean) {
  id += 1;
  return { id, date, people, salary, isFree, isPersonal, isJumps };
}

const rows = [
  createData(new Date('December 17, 1995 03:24:00'), 222, 222222, false, true, false),
];

const Table: React.FC<TableProps> = ({}) => {
  return (
    <MaterialPaper>
      <MaterialTable>
        <TableHead>
          <TableRow>
            <TableCell>Дата</TableCell>
            <TableCell>Кол-во человек</TableCell>
            <TableCell>Прибыль</TableCell>
            <TableCell>Бесплатная</TableCell>
            <TableCell>Персональная</TableCell>
            <TableCell>Джампы</TableCell>
          </TableRow>
        </TableHead>

        <TableBody>
          {rows.map((row) => (
            <TableRow key={row.id}>
              <TableCell component='th' scope='row'>
                {`${row.date.getDate()}.${row.date.getMonth()}`}
              </TableCell>
              <TableCell align='left'>{row.people}</TableCell>
              <TableCell align='left'>{row.salary}</TableCell>
              <TableCell align='left'>{row.isFree && <Done />}</TableCell>
              <TableCell align='left'>{row.isPersonal && <Done />}</TableCell>
              <TableCell align='left'>{row.isJumps && <Done />}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </MaterialTable>
    </MaterialPaper>
  );
};

export default memo(Table);
