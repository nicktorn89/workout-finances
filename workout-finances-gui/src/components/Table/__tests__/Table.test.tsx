import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Table from '../Table';

describe('<Slider />', () => {
  const TableComponent = shallow(<Table data={[]} />);
  const TableComponentToJson = shallowToJson(TableComponent);
  
  it('Should be equal to snapshot', () => {
    expect(TableComponentToJson).toMatchSnapshot();
  });
});
