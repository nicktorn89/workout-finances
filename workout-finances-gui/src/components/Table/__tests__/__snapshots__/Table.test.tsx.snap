// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Should be equal to snapshot 1`] = `
<Styled(WithStyles(Paper))>
  <Styled(WithStyles(Table))>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)>
          Дата
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Кол-во человек
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Прибыль
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Бесплатная
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Персональная
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Джампы
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)
        key="1"
      >
        <WithStyles(TableCell)
          component="th"
          scope="row"
        >
          17.11
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          align="left"
        >
          222
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          align="left"
        >
          222222
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          align="left"
        />
        <WithStyles(TableCell)
          align="left"
        >
          <pure(DoneIcon) />
        </WithStyles(TableCell)>
        <WithStyles(TableCell)
          align="left"
        />
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </Styled(WithStyles(Table))>
</Styled(WithStyles(Paper))>
`;
